### PR TITLE
BDD Front components

### PR DESCRIPTION
**Pull request by Gauthier-Lhomme**

### CODE REVIEW

**Review by Marie-Garde — DISMISSED** [review]
Let's do this Loic !

**Review by Marie-Garde — DISMISSED** [review]
Let's do this Loic !

**Comment by LoicBrassart** [comment]
Yeah, let's do this !! :triumph: 

(Seriously, what's happening ? :laughing: : )

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@testing-library/react": "^11.2.2",
     "@testing-library/user-event": "^12.5.0",
     "axios": "^0.21.1",
+    "emailjs-com": "^2.6.4",
```
Comment (LoicBrassart) [inline package.json]: What's this package ?

File: src/App.js [hunk 2]
```diff
@@ -10,17 +13,20 @@ import Sommaire from "./components/header/HeaderMenu00";
...
 import Inscription from "./components/Inscription";
 import DisplayClient from "./components/DisplayClient";
-import ActionPage from "./components/ActionPage";
-import { Switch, Route } from "react-router-dom";
 import RessourcePage from "./components/RessourcePage";
-import MainPage from "./components/Main";
 
 export default function App() {
+  const [token, setToken] = useState("");
+  console.log(token);
```
Comment (LoicBrassart) [inline src/App.js]: No `console.log`

File: src/App.js [hunk 3]
```diff
@@ -10,17 +13,20 @@ import Sommaire from "./components/header/HeaderMenu00";
...
       <Reset />
       <HeaderTop />
       <Switch>
+        <Route path="/clientpage" component={ClientPage} />
+        <Route exact path="/">
+          <Connexion setToken={setToken} />
+        </Route>
         <Route exact path="/" component={MainPage} />
         <Route exact path="/sommaire" component={Sommaire} />
         <Route exact path="/" component={Mainpage} />
```
Comment (LoicBrassart) [inline src/App.js]: Three of your `Routes` take the same path, you'll never load `MainPage`

File: src/components/ContactUs.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        e.target,
+        "user_NKYLcqZwiFms9AL3RjBZO"
+      )
+      .then(
+        (result) => {
+          console.log(result.text);
+        },
+        (error) => {
+          console.log(error.text);
+        }
```
Comment (LoicBrassart) [inline src/components/ContactUs.jsx]: No `console.log`

File: src/components/InscriptionSignUp.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    setErrors(validateInfo(values));
+    delete values.password2;
+    axios
+      .post("http://localhost:3001/auth/signup", values)
+      .then((res) => {
+        setValues(res.data);
+        history.push("/clientpage");
+      })
+      .catch((e) => {
+        console.warn(e);
```
Comment (LoicBrassart) [inline src/components/InscriptionSignUp.jsx:43]: Nope, still no `console.log`

File: src/components/ProfilPage.jsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import ContactUs from "./ContactUs.jsx";
+
+export default () => {
+  const [clients, setClients] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    axios.get("http://localhost:3001/client/1").then((res) => {
+      console.log(res.data);
```
Comment (LoicBrassart) [inline src/components/ProfilPage.jsx]: Nooooooooooope

File: src/components/ProfilPage.jsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        <Text>
+          <Name>Name</Name>
+          <Job>Function</Job>
+          <Company>Company</Company>
+        </Text>
+      </FlexBox1>
+      <FlexBox2>
+        <div>
+          {loading && <p>it's loading</p>}
+          {!loading && (
```
Comment (LoicBrassart) [inline src/components/ProfilPage.jsx]: Use a ternary:
```
{loading ? <p>it's loading</p> : {*monstro-code goez here*}}

File: src/components/Sidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import axios from "axios";
+
+export default function Sidebar() {
+  const [clients, setClients] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    axios.get("http://localhost:3001/client/1").then((res) => {
+      console.log(res.data);
```
Comment (LoicBrassart) [inline src/components/Sidebar.jsx]: Ah ? ... Nope

File: src/components/ValidateInfo.jsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+export default function validateInfo(values) {
+  let errors = {};
+  if (!values.password) {
+    errors.password = "Un mot de passe est requis";
+  } else if (values.password.lenght < 8) {
+    errors.password = "Le mot de passe est trop court (8 charactéres minimums)";
```
Comment (LoicBrassart) [inline src/components/ValidateInfo.jsx]: Typo: "caractères"

File: src/components/ValidateInfo.jsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+export default function validateInfo(values) {
+  let errors = {};
+  if (!values.password) {
+    errors.password = "Un mot de passe est requis";
+  } else if (values.password.lenght < 8) {
```
Comment (LoicBrassart) [inline src/components/ValidateInfo.jsx]: Typo (this never works !! :warning: ) : `password.length`

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
Just no, too many Evils v_v

**Comment by LoicBrassart** [comment]
I'd like to add that I'm sure nobody proofread this PR. 
It's refused because of "basic" errors, and they should have been spotted by anyone really reading the code

File: package.json (hunk 1 above)
Comment (Marie-Garde) [reply package.json]: Sending an email

File: src/components/ProfilPage.jsx (hunk 6 above)
Comment (Marie-Garde) [reply src/components/ProfilPage.jsx]: Sorry T.T

**Review by Thoms-C — DISMISSED** [review]
Seems ok to me !

**Review by Thoms-C — DISMISSED** [review]
Seems pretty ok

**Review by Thoms-C — APPROVED** [review]
Ok !

**Review by LoicBrassart — APPROVED** [review]
NmE at last!

**Pull request merged**